lib/roo_on_rails/routemaster/publisher.rb

```diff
@@ -0,0 +1,49 @@
+module RooOnRails
+  module Routemaster
+    class Publisher
+      attr_reader :model, :event
+
+      def initialize(model, event)
+        @model = model
+        @event = event
+      end
+
+      def publish?
+        true
+      end
+
+      def publish!
+        return unless publish?
+
+        event_data = data # cache in case it isn't memoised
```

Comment:
To be fair this pattern is copied from your code ;-)

But yes I did mean memoise. It could be cleaned by handling nil in the stringify method though.